.github/workflows/release.yml

```diff
@@ -0,0 +1,139 @@
+        sources="$(\
+          hatch run default:pip list --verbose --format json \
+          | jq -r '.[] | select(.editable_project_location == null) | "\(.name);\(.location)"' \
+        )"
+        echo "::group::Dependencies"
+        printf '%s\n' "${sources}"
+        echo "::endgroup::"
+        mkdir bundle/
+        cp -r yt_dlp_plugins bundle/
+        while IFS=';' read -r name path; do
+          if [[ ! "${name}" =~ ^(pip|setuptools|wheel|yt-dlp)$ ]]; then
+            package_name="$(tr '[:upper:]' '[:lower:]' <<<"${name}" | sed 's/-/_/g')"
+            cp -r "${path}/${package_name}" bundle/
+          fi
+        done <<<"${sources}"
+        cd bundle/
+        find . -type f -name '*.py[co]' -delete -o -type d -name __pycache__ -delete
+        zip -9 --recurse-paths "${project_name}" *
+        gh release create "${version}" --latest \
+          --title "${project_name} ${version}" \
+          "${project_name}.zip"
+
+  release_pypi:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: "3.10"
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build hatchling
+
+    - name: Build
+      run: |
+        cp README.md plugin/
+        cd plugin/
+        rm -rf dist/**
+        printf '%s\n\n' \
+          "Official repository: <${{ github.server_url }}/${{ github.repository }}>" > ./README.md.new
+        cat ./README.md >> ./README.md.new && mv -f ./README.md.new ./README.md
+        python -m build --no-isolation .
+
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        verbose: true
+        packages-dir: plugin/dist
+
+  release_docker:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        context: git
+        images: ${{ github.repository }}  # like `brainicism/bgutil-ytdlp-pot-provider`
+        tags: |
+          type=raw,value=latest,enable={{is_default_branch}}
+          type=semver,pattern={{version}}
+          type=sha
+          type=raw,value=ci
+
+    - name: Login to DockerHub
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}  # set the secret
+        password: ${{ secrets.DOCKERHUB_TOKEN }}  # set the secret
+
+    - name: Build and push
+      uses: docker/build-push-action@v6
+      with:
+        context: server/
+        file: server/Dockerfile
+        push: ${{ github.event_name != 'pull_request' }}
```

Comment:
There's a couple of this `event` checks, but this workflow is currently allowed for `workflow_dispatch`. Are you anticipating integrating this into the PR workflow in the future?

Reply:
No. Just copied from some document. This actually replaces that build.yml if we change the workflow `on`